### PR TITLE
Improve user creation listeners to validate username format.

### PR DESCRIPTION
**Pull request by Thisara-Welmilla**
Improve the user creation listeners to validate against the validation configuration for username in exists.

### CODE REVIEW

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/listener/InputValidationListener.java [hunk 1]
```diff
@@ -110,62 +118,75 @@ public boolean doPreUpdateCredentialWithID(String userID, Object newCredential,
...
+                        if (PASSWORD.equals(field)) {
+                            UserCoreUtil.setSkipPasswordPatternValidationThreadLocal(true);
+                        } else if (USERNAME.equals(field)) {
+                            UserCoreUtil.setSkipUsernamePatternValidationThreadLocal(true);
+                        }
+                        validateAgainstConfiguration(configuration, validators, field, fieldValueMap.get(field),
+                                tenantDomain);
+                    } catch (InputValidationMgtClientException e) {
+                        if (LOG.isDebugEnabled()) {
+                            LOG.debug(new StringFormattedMessage("Failed to validate %s for user. " +
```
Comment (DMHP) [inline components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/listener/InputValidationListener.java:159]: Let's add an error log

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/listener/InputValidationListener.java [hunk 2]
```diff
@@ -71,10 +75,14 @@ public boolean doPreAddUser(String userName, Object credential, String[] roleLis
         if (!isEnable()) {
             return true;
         }
-        if (UserCoreUtil.getSkipPasswordPatternValidationThreadLocal()) {
-            return true;
+        Map<String, String> validationRequiredField = new HashMap<>();
```
Comment (DMHP) [inline components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/listener/InputValidationListener.java:78]: validationRequiredField > add a meaningful name and comments explaining the usage of the map

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/listener/InputValidationListener.java [hunk 3]
```diff
@@ -110,62 +118,75 @@ public boolean doPreUpdateCredentialWithID(String userID, Object newCredential,
...
+        return validate(Collections.singletonMap(PASSWORD, newCredential.toString()), userStoreManager);
 
     }
 
     /**
      * Method to validate the values.
      *
-     * @param field             Field to be validated.
-     * @param value             Value to be validated.
+     * @param fieldValueMap     Map of fields and values that need to be validated.
```
Comment (DMHP) [inline components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/listener/InputValidationListener.java:128]: Add a sample value and meaningful map name

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/listener/InputValidationListener.java [hunk 4]
```diff
@@ -110,62 +118,75 @@ public boolean doPreUpdateCredentialWithID(String userID, Object newCredential,
...
+                ValidationConfiguration configuration = configurations.stream().filter(config ->
+                        field.equalsIgnoreCase(config.getField())).collect(Collectors.toList()).get(0);
+                if (configuration != null) {
+                    try {
+                        if (PASSWORD.equals(field)) {
+                            UserCoreUtil.setSkipPasswordPatternValidationThreadLocal(true);
+                        } else if (USERNAME.equals(field)) {
+                            UserCoreUtil.setSkipUsernamePatternValidationThreadLocal(true);
+                        }
+                        validateAgainstConfiguration(configuration, validators, field, fieldValueMap.get(field),
```
[the remark this answers is not in the document]
Comment (Thisara-Welmilla) [reply components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/listener/InputValidationListener.java]: Addressed

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/listener/InputValidationListener.java (hunk 2 above)
Comment (Thisara-Welmilla) [reply components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/listener/InputValidationListener.java]: Addressed

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/listener/InputValidationListener.java (hunk 3 above)
Comment (Thisara-Welmilla) [reply components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/listener/InputValidationListener.java]: Addressed

File: components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/listener/InputValidationListener.java (hunk 1 above)
Comment (Thisara-Welmilla) [reply components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/listener/InputValidationListener.java]: Addressed

**Review by DMHP — APPROVED** [review]

**Pull request merged**